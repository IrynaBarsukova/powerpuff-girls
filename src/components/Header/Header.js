import React from 'react';
import { Link } from 'react-router-dom';

import Container from '../common/Container/Container';
import logo from '../../assets/images/the-powerpuff-girls.png';
import styles from './Header.scss';

function Header() {
    return (
        <header className={styles.header}>
            <Container>
                <Link to="/" className={styles.header__link}>
                    <h1 className={styles.header__title}>The Powerpuff Girls</h1>
                    <img className={styles.header__logo} src={logo} alt="Logo"/>
                </Link>
            </Container>
        </header>
    );
}

export default Header;
