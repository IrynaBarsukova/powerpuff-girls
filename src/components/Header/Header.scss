.header {
  padding: 17px 0;
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-color: #f571b4;

  &__title {
    visibility: hidden;
    font-size: 0;
    color: #607d8b;
    margin: 0;
  }
}

.header__link {
  align-self: flex-start;
  text-decoration: none;
}

.header__logo {
  width: 300px;
}
