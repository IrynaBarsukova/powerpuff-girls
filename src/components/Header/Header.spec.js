import React from 'react';

import renderWithRouter from '../../mocks/router.mock';
import Header from './Header';

describe('<Header />', () => {
    it('should render Header', () => {
        const { container } = renderWithRouter(<Header />);

        expect(container).toMatchSnapshot();
    });

    it('should navigate to Home when the logo is clicked', () => {
        const { getByText } = renderWithRouter(<Header />);

        expect(getByText('The Powerpuff Girls').closest('a')).toHaveAttribute(
            'href',
            '/',
        );
    });
});
