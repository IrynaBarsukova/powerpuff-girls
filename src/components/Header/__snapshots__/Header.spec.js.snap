// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <a
        class="header__link"
        href="/"
      >
        <h1
          class="header__title"
        >
          The Powerpuff Girls
        </h1>
        <img
          alt="Logo"
          class="header__logo"
          src="[object Object]"
        />
      </a>
    </div>
  </header>
</div>
`;
